src/main/java/org/apache/commons/imaging/formats/jpeg/decoder/JpegDecoder.java

```diff
@@ -38,7 +41,7 @@
 
 import static org.apache.commons.imaging.common.BinaryFunctions.*;
 
-public class JpegDecoder extends BinaryFileParser implements JpegUtils.Visitor {
+public final class JpegDecoder extends BinaryFileParser implements JpegUtils.Visitor {
```

Comment:
Not necessary to address this here. We can discuss and add the final later if necessary :-)